Add comment_count property to returned review object

callumu21 opened the pull request:
**GET /api/review/:review_id**

Added an extra property on the returned review (comment_count) and expanded the database query to return the correct number of comments using an outer join.

callumu21 committed:
Add comment_count property to returned review object

daviidbartlett commented on __tests__/app.test.js:
```diff
@@ -50,7 +50,7 @@ describe("/api/categories", () => {
 
 describe("/api/reviews/:review_id", () => {
-  describe("GET", () => {
+  describe.only("GET", () => {
```
remove .only

daviidbartlett commented on __tests__/app.test.js:
```diff
@@ -70,10 +70,25 @@ describe("/api/reviews/:review_id", () => {
               category: expect.any(String),
               owner: expect.any(String),
               created_at: expect.any(String),
+              comment_count: expect.any(Number),
```
you shouldn't be editing previous tests for new functionality. I'd expect this to be a new test.

daviidbartlett replied:
if you have you change an old test it's a clue that it was a brittle test

daviidbartlett commented on models/models.js:
```diff
@@ -8,8 +8,12 @@ exports.fetchCategories = () => {
   return db
-    .query("SELECT * FROM reviews WHERE review_id = $1", [review_id])
+    .query(
+      "SELECT reviews.*, COUNT(comments.review_id)::int AS comment_count FROM reviews LEFT OUTER JOIN comments ON comments.review_id = reviews.review_id WHERE reviews.review_id = $1 GROUP BY reviews.review_id",
```
nice, might be worth using ```` to so you can wrap the string.

daviidbartlett commented on models/models.js:
```diff
@@ -8,8 +8,12 @@ exports.fetchCategories = () => {
+      [review_id]
+    )
     .then(({ rows: review }) => {
+      console.log(review);
```
gross console.log KILL IT WITH FIRE

callumu21 commented:
Removed the console.log from models.js and the .only from the test block. I've also refactored the tests so that the additional property is its own test and included backticks to reformat the query into a neater display.
